compiler/rustc_arena/src/lib.rs

```diff
@@ -217,8 +237,10 @@ impl<T> TypedArena<T> {
             let mut chunks = self.chunks.borrow_mut();
             let mut new_cap;
             if let Some(last_chunk) = chunks.last_mut() {
-                let used_bytes = self.ptr.get() as usize - last_chunk.start() as usize;
-                last_chunk.entries = used_bytes / mem::size_of::<T>();
+                if mem::needs_drop::<T>() {
```

Comment:
Why is this needed?

Reply:
There are some places where TypedArena is created for non-drop types. It doesn't need to track the entries in that case.

Reply:
That seems like an orthogonal change to this PR that I'd be more comfortable reviewing in a separate PR. Especially since I think there are more places to do this then.

Reply:
I only found this one, but I can extract it into a different PR